### PR TITLE
refactor: usage of execution details status to job status

### PR DESCRIPTION
**Pull request by davidsoderberg**
### What kind of change does this PR introduce? 
- [ ] Bug
- [x] Feature
- [ ] Docs
- [ ] Other(s)

### CODE REVIEW

**Review by ainouzgali — APPROVED** [review]
⭐️

File: apps/api/src/app/events/usecases/add-job/add-job.usecase.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class AddJob {
 
     const delay = digestAmount ?? delayAmount;
 
-    this.createExecutionDetails.execute(
+    await this.createExecutionDetails.execute(
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/add-job/add-job.usecase.ts:44]: Good one!

File: apps/web/src/components/activity/helpers.ts [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
-  if (status === ExecutionDetailsStatusEnum.SUCCESS) {
-    return CheckCircle;
-  }
-
-  if (status === ExecutionDetailsStatusEnum.FAILED) {
-    return ErrorIcon;
-  }
-
-  return Timer;
+export const getLogoByStatus = (): React.FunctionComponent<React.ComponentPropsWithoutRef<'svg'>> => {
```
Comment (p-fernandez) [inline apps/web/src/components/activity/helpers.ts:19]:
```suggestion
export const getCheckLogo = (): React.FunctionComponent<React.ComponentPropsWithoutRef<'svg'>> => {
```

File: apps/web/src/components/activity/ExecutionDetail.tsx [hunk 3]
```diff
@@ -26,14 +26,14 @@ const ExecutionDetailTime = ({ createdAt }) => {
...
-const getExecutionDetailStatus = (status, type) => {
+const getExecutionDetailStatus = (type) => {
   const logo = getLogoByType(type);
 
   if (logo) {
     return logo;
   }
 
-  return getLogoByStatus(status);
+  return getLogoByStatus();
```
Comment (p-fernandez) [inline apps/web/src/components/activity/ExecutionDetail.tsx:36]:
```suggestion
  return getCheckLogo();
```

File: apps/web/src/components/activity/ExecutionDetail.tsx [hunk 4]
```diff
@@ -26,14 +26,14 @@ const ExecutionDetailTime = ({ createdAt }) => {
   return <FormattedTime>{formattedTime}</FormattedTime>;
 };
 
-const getExecutionDetailStatus = (status, type) => {
+const getExecutionDetailStatus = (type) => {
```
Comment (p-fernandez) [inline apps/web/src/components/activity/ExecutionDetail.tsx:29]:
```suggestion
const getExecutionDetailLogoByType = (type) => {
```

File: apps/web/src/components/activity/ExecutionDetail.tsx [hunk 5]
```diff
@@ -45,7 +45,7 @@ export const ExecutionDetail = ({
   const theme = useMantineTheme();
   const { createdAt, detail, raw, status, type } = executionDetail;
   const color = getColorByStatus(theme, status);
-  const ExecutionDetailStatus = getExecutionDetailStatus(status, type);
+  const ExecutionDetailStatus = getExecutionDetailStatus(status);
```
Comment (p-fernandez) [inline apps/web/src/components/activity/ExecutionDetail.tsx:48]:
```suggestion
  const ExecutionDetailLogo = getExecutionDetailLogoByType(type);
```

**Review by p-fernandez — COMMENTED** [review]
You are so quick! ⚡ 
Some changes needed to make fully sense the overall change, in the function helpers that render the logos for the execution details.

**Comment by scopsy** [comment]
@p-fernandez we actually ended up using the `Job` existing status as it was mapping exactly for what we needed it to be ^

File: apps/api/src/app/events/usecases/send-message/send-message-email.usecase.ts [hunk 6]
```diff
@@ -310,9 +316,10 @@ export class SendMessageEmail extends SendMessageType {
         })
       );
 
-      if (!result.id) {
+      if (!result?.id) {
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/send-message/send-message-email.usecase.ts:319]: This broke a test. But that also means that sending the mail doesn't return anything anymore. Worth to double check.

File: apps/api/src/app/events/usecases/send-message/send-message-sms.usecase.ts [hunk 7]
```diff
@@ -299,9 +306,10 @@ export class SendMessageSms extends SendMessageType {
         })
       );
 
-      if (!result.id) {
+      if (!result?.id) {
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/send-message/send-message-sms.usecase.ts:309]: This little one fixed another test.

**Pull request merged**